src/librustc_codegen_ssa/mir/block.rs

```diff
@@ -544,15 +544,42 @@ impl<'a, 'tcx, Bx: BuilderMethods<'a, 'tcx>> FunctionCx<'a, 'tcx, Bx> {
         }
 
         // Emit a panic or a no-op for `panic_if_uninhabited`.
-        if intrinsic == Some("panic_if_uninhabited") {
+        // These are intrinsics that compile to panics so that we can get a message
```

Comment:
Would be good if we could refactor into more methods to improve readability here.